cli/cmd/root.go

```diff
@@ -150,7 +150,7 @@ func init() {
 	rootCmd.PersistentFlags().StringVar(&server, "server", "", "VMClarity server to export scan results to, for example: http://localhost:9999/api")
 	rootCmd.PersistentFlags().StringVar(&scanResultID, "scan-result-id", "", "the ScanResult ID to export the scan results to")
 	rootCmd.PersistentFlags().BoolVar(&mountVolume, "mount-attached-volume", false, "discover for an attached volume and mount it before the scan")
-	rootCmd.PersistentFlags().BoolVar(&waitForServerAttached, "wait-for-server-attached", false, "wait for the VMClarity server to attach the volume")
+	rootCmd.PersistentFlags().BoolVar(&waitForReadyState, "wait-for-ready", false, "wait until the ScanResult state is set to ReadyToScan")
```

Comment:
Do we need this flag? When we will not want to wait for scan to be ready to be scanned?

Reply:
we don't want to wait when the CLI is standalone without server, and CICD case when there is no ScanResult to wait on.

 Perhaps we decide to WaitForReady based on being given the Server ID and ScanResult ID?

Reply:
Both the `ScanResult` and `ReadyToScan` are part of the public API, so I would not describe them as internal. I think this help the user to understand better the use of this flag. Its kinda the same as the `--scan-result-id` flag.
But I have no preference on this one. I am good with changing the help string to your suggestion.

Reply:
> Perhaps we decide to WaitForReady based on being given the Server ID and ScanResult ID?

This could work as well. I think this is the way to go.

Reply:
> This could work as well. I think this is the way to go.

So in another PR?

Reply:
Nope, I'll update the PR with the suggestion. :)

Reply:
Done. The `WaitForReadyState` will be called always and the `state.Manager` implementation will decide whether it needs to actually wait until the state is set to `ReadyToScan` or just skips it. I think it is better to delegate this to the  state manager.

Reply:
where is the code that the manager decide upon? or it is just because `state.NewLocalState()` is used when `server` is empty? I hope it works well for the cicd mode.